pyclean/modern.py

```diff
@@ -400,3 +404,101 @@ def suggest_debris_option(args):
             'Hint: Use --debris to also clean up build artifacts '
             'from common Python development tools.',
         )
+
+
+def build_git_clean_command(ignore_patterns, dry_run=False, force=False):
+    """
+    Build the git clean command with appropriate flags.
+
+    Args:
+        ignore_patterns: List of patterns to exclude from git clean
+        dry_run: If True, only show what would be deleted
+        force: If True, delete without prompting (otherwise interactive)
+
+    Returns:
+        List of command arguments
+    """
+    cmd = ['git', 'clean', '-dx']
+
+    for pattern in ignore_patterns:
+        cmd.extend(['-e', pattern])
+
+    if dry_run:
+        cmd.append('-n')
+    elif force:
+        cmd.append('-f')
+    else:
+        cmd.append('-i')
+
+    return cmd
+
+
+def execute_git_clean(directory, args):
+    """
+    Execute git clean in the specified directory.
+
+    Args:
+        directory: Path to the directory to clean
+        args: Command line arguments with ignore, dry_run, yes flags
+    """
+    cmd = build_git_clean_command(
+        args.ignore,
+        dry_run=args.dry_run,
+        force=args.yes
+    )
+
+    log.debug('Running: %s in %s', ' '.join(cmd), directory)
+
+    result = subprocess.run(
+        cmd,
+        cwd=directory,
+        capture_output=True,
+        text=True,
+        check=False,
+    )
+
+    if result.stdout:
+        log.info(result.stdout.rstrip())
+    if result.stderr:
+        log.warning(result.stderr.rstrip())
+
+    if result.returncode == 128:
+        log.warning(
+            'Directory %s is not under version control. Skipping git clean.',
+            directory,
+        )
+    elif result.returncode != 0:
```

Comment:
```suggestion
    elif result.returncode:
```

... would be more pythonic.